architecture-evaluation/src/main/java/edu/kit/kastel/sdq/case4lang/refactorlizar/architecture_evaluation/graphs/SpoonNode.java

```diff
@@ -0,0 +1,53 @@
+package edu.kit.kastel.sdq.case4lang.refactorlizar.architecture_evaluation.graphs;
+
+import java.util.Objects;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.declaration.CtTypeMember;
+import spoon.reflect.visitor.CtVisitor;
+
+public class SpoonNode implements Node<CtType<?>> {
+    private final CtTypeMember member;
+
+    /** @param member */
+    public SpoonNode(CtTypeMember member) {
+        this.member = member;
+    }
+
+    /* (non-Javadoc)
+     * @see java.lang.Object#hashCode()
+     */
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(member, member.getDeclaringType());
+    }
+
+    /* (non-Javadoc)
+     * @see java.lang.Object#equals(java.lang.Object)
+     */
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (!(obj instanceof SpoonNode)) return false;
+        SpoonNode other = (SpoonNode) obj;
+        return Objects.equals(member, other.member)
+                && Objects.equals(member.getDeclaringType(), other.member.getDeclaringType());
+    }
+
+    /**
+     * @param arg0
+     * @see spoon.reflect.visitor.CtVisitable#accept(spoon.reflect.visitor.CtVisitor)
+     */
+    public void accept(CtVisitor arg0) {
+        member.accept(arg0);
+    }
+
+    /**
+     * @return
+     * @see spoon.reflect.declaration.CtTypeMember#getDeclaringType()
+     */
+    public CtType<?> getModule() {
+        return member.getDeclaringType();
+    }
```

Comment:
[EmptyBlockTag] A block tag (@param, @return, @throws, @deprecated) has an empty description. Block tags without descriptions don't add much value for future readers of the code; consider removing the tag entirely or adding a description.

[MissingSummary] A summary fragment is required; consider using the value of the @see block as a summary fragment instead.

[MissingOverride] getModule implements method in Node; expected @Override
    public CtType<?> getModule() {

Reply:
Also I did not change this method, just renamed it. Anyway, added a comment.

Reply:
Sadly, there were already some error-prone smells, I tried to filter them. We still try to reduce technical debt, there is not always time left for it. Thanks for the comment and sorry for the inconvenience.